From patchwork Wed Mar 24 17:50:02 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Krish Sadhukhan <krish.sadhukhan@oracle.com>
X-Patchwork-Id: 12162017
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,MSGID_FROM_MTA_HEADER,SPF_HELO_NONE,
	SPF_PASS,USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 2AC47C43381
	for <kvm@archiver.kernel.org>; Wed, 24 Mar 2021 18:39:34 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 1341E61A26
	for <kvm@archiver.kernel.org>; Wed, 24 Mar 2021 18:39:34 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S237736AbhCXSjD (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Wed, 24 Mar 2021 14:39:03 -0400
Received: from aserp2120.oracle.com ([141.146.126.78]:49182 "EHLO
        aserp2120.oracle.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S237687AbhCXSis (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 24 Mar 2021 14:38:48 -0400
Received: from pps.filterd (aserp2120.oracle.com [127.0.0.1])
        by aserp2120.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 12OIPba9047435;
        Wed, 24 Mar 2021 18:38:44 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : in-reply-to : references :
 content-transfer-encoding : content-type : mime-version;
 s=corp-2020-01-29; bh=ovKKeLIITNnOyf7sYV7ejNLNQPh7Qh+PcU7ZNzLbbR4=;
 b=ROVeNDiMxBbMZPDtLIWfujOxX1OGdIwF+LaEaBerhAx55Dlq78M+5qXtSBSrVnkPMglP
 +BZFfkU8cHeS9WGXyzwii5r0blkXmohuwGRi6GLVEm9aocVDDWYXBYk4Ah2ER9Jxe3Oj
 SowmjDv8NamwzDyuB5XqbEmwgGvtosS1RUYXnP6DyVsBvR5YmA2QlH9Kjblfd3eKQk2l
 MiANHRxhHwxYgsv+FYFOUZ98HHDomJYqmgL4pDbApKcqpA2zDyM1hKxhBqcOFUThgagm
 D3VHtV/KmENqaLI7WEJlz2ck49yENApDhKwR2p9hDKJ4OsZZTbrlr3Z1HJIB6C8uh0oS 8A==
Received: from userp3020.oracle.com (userp3020.oracle.com [156.151.31.79])
        by aserp2120.oracle.com with ESMTP id 37d90mktnv-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Wed, 24 Mar 2021 18:38:44 +0000
Received: from pps.filterd (userp3020.oracle.com [127.0.0.1])
        by userp3020.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 12OIQFYJ167607;
        Wed, 24 Mar 2021 18:38:43 GMT
Received: from nam12-mw2-obe.outbound.protection.outlook.com
 (mail-mw2nam12lp2040.outbound.protection.outlook.com [104.47.66.40])
        by userp3020.oracle.com with ESMTP id 37dtttpj5x-2
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Wed, 24 Mar 2021 18:38:43 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=kZVNV7PECT23hhKAItgpOLUZ+WczJ6Zgrt4wVpTRxU0cr5LUIRXY5PxWSKfZRTZACl8U5CQwTcxpan1PBLmhvRuaLDX6wI0ezuyTofuf6Kh4uD+UUNVQutqZsFA6uAk5LBY7lncIt542yTJWWFgl0sXyaV4wQ4LzEhaIz/yF17aPGCYbBQuZ85yo7gl7wmw0E71P2+B4ZzSJM3x8oWPINyW1pJNFr+E+GioTPJS+bjC5qQsg9D5MEun1hfgCXTTJudDTZPi5c7jqHybaGIrbJP/yurFxLuxcjtlBDoIoJSymUWdBrPS5OcDpigYr6/JLThv2AMVWPl0DyduV7Pex8A==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=ovKKeLIITNnOyf7sYV7ejNLNQPh7Qh+PcU7ZNzLbbR4=;
 b=hiEKPEBRe4Vhcikd1xI9NIAj8sO/BUUerrHDY0ijQkAYPaTeoqnCdVi/nC1cTuUvgQYb8x8UrhNTlXSP/vTjX+KqWdVqalrk4/nW/mZaz9lVpzIoc/Epoq73KIBArlr4NYtW8wP0NQ/ZRUWFFX8pau06feFeXKN4szi3JtHHTElOI4SdR8n4nShX40g7O/2hfAU5sSjr2bgJjT6mSU57/OlgA8cIPEZnsMsua0GR90CnZehcLNM949kikGejpAumKve3i6uFucASTsh32cBsNO8mLpKF7R2uEeZfYSJ11p16SJESMYOSd2zDD3aFEqR2iv0BK3k9/KPOWizRbY82vg==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=ovKKeLIITNnOyf7sYV7ejNLNQPh7Qh+PcU7ZNzLbbR4=;
 b=MZ+wIi3sNi0qbgFYYdFnRYljWfc922zvU1q6mU0QlXkmApXrVIVDAKJ8GVZAVj/b3KZRMpw6zQP6dkh/Y5NWOwrIDa8QBKrv6oIp7Uks/x9jqaLJUQpBUeVV5ZQc72K0IoJWFVHyJa7baEW0J4axIsYbc4um1gMzNBsvnhcnM70=
Authentication-Results: vger.kernel.org; dkim=none (message not signed)
 header.d=none;vger.kernel.org; dmarc=none action=none header.from=oracle.com;
Received: from SN6PR10MB3021.namprd10.prod.outlook.com (2603:10b6:805:cc::19)
 by SA2PR10MB4665.namprd10.prod.outlook.com (2603:10b6:806:fb::17) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3955.18; Wed, 24 Mar
 2021 18:38:42 +0000
Received: from SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::1871:3741:cc17:bcf7]) by SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::1871:3741:cc17:bcf7%7]) with mapi id 15.20.3955.025; Wed, 24 Mar 2021
 18:38:42 +0000
From: Krish Sadhukhan <krish.sadhukhan@oracle.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, jmattson@google.com, seanjc@google.com
Subject: [PATCH 1/5 v4] KVM: SVM: Move IOPM_ALLOC_ORDER and MSRPM_ALLOC_ORDER
 #defines to svm.h
Date: Wed, 24 Mar 2021 13:50:02 -0400
Message-Id: <20210324175006.75054-2-krish.sadhukhan@oracle.com>
X-Mailer: git-send-email 2.25.4
In-Reply-To: <20210324175006.75054-1-krish.sadhukhan@oracle.com>
References: <20210324175006.75054-1-krish.sadhukhan@oracle.com>
X-Originating-IP: [138.3.201.29]
X-ClientProxiedBy: BYAPR05CA0009.namprd05.prod.outlook.com
 (2603:10b6:a03:c0::22) To SN6PR10MB3021.namprd10.prod.outlook.com
 (2603:10b6:805:cc::19)
MIME-Version: 1.0
X-MS-Exchange-MessageSentRepresentingType: 1
Received: from ban25x6uut29.us.oracle.com (138.3.201.29) by
 BYAPR05CA0009.namprd05.prod.outlook.com (2603:10b6:a03:c0::22) with Microsoft
 SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3977.10 via Frontend
 Transport; Wed, 24 Mar 2021 18:38:41 +0000
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 6fe74b40-3715-4ee5-38aa-08d8eef40c54
X-MS-TrafficTypeDiagnostic: SA2PR10MB4665:
X-Microsoft-Antispam-PRVS: 
 <SA2PR10MB4665D614EB1AAE5160DFDA5981639@SA2PR10MB4665.namprd10.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:3513;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 f5VSKIaVyoOfClAgEn+jaf65a9qO/MC1aNoYrUiJiueG+WbFJ3xjaJtnfWA1VfsucmyFrW94JvbERcHN8ZezaXDPmK+AUL5I+CcARDGF0hyV0kBzvFr04vL45Al79dkK7QOcjdVv1arGU0uj3dknzmHBKdGQy1QbsuUEYBpT/X4QZjQWWKcmc4JW8yzdZkPYtEmOe/gN3+7MyMtOgETgdlIOnMVizzBUwdA3rTmEYusVp1ZKoIwmWJw/gmYFQWzWzD1vJ7Gaikkt7BUfXleKdXmculWIfPLk/iiwZETrUwm/rM4a3OThsEDrVoHu0AQvmUiptGKZSd3cFVg1XO+E6TXBrQYV5KNh/Nl9Eto5zxPGFVXeqULP4O5NcD+44QDS3Qu9wcF1KE+R+lbubUhSwWNkd/g1ZK2WmVE2C3K1djHcYOFVN4XMtMIOHHw9ORxXHxHqZpbDzWFA9Ufddb5F0M1oLN/SbtTDfzqz9HPZJHRY1jeK8EGA/HqD1n6wr+wkwmH/VnikK/7Su7lTeq8zl5MJ7SU4n+87rJI3drlEVhBfixt/LxnryCkCMX4gzlQku4T3ctmd+Ym9C7wt6Yh/6bmOHARq2X6r63yGgNuY3d5GtAo0Dqwsrah5Ihff1euKY4v0RyMuroAyLUQ7MM4qxw==
X-Forefront-Antispam-Report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3021.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(376002)(346002)(396003)(366004)(39860400002)(136003)(66946007)(6486002)(36756003)(16526019)(186003)(478600001)(6666004)(956004)(1076003)(6916009)(2616005)(8676002)(38100700001)(4326008)(83380400001)(44832011)(316002)(66476007)(66556008)(5660300002)(2906002)(7696005)(52116002)(8936002)(86362001)(26005);DIR:OUT;SFP:1101;
X-MS-Exchange-AntiSpam-MessageData: 
 zrdPEycOCvUJ1M6TgcI+Rfn5vK3odD9tRwiexNUGMr/vg3kcyN6OfCqWuXL2E/Mmhwamad5oWpG2XAQ3WMecWgsNNZyhuUaa7azPDCk0Hx7Vujr4mj89Rc+8HRfFwcD65bXVngxzuPYDcWWTOTNYv1U4dtcpxC5LqDLiTXydj3fPt/ubBMClMF30mRvkXOHb7UPCWa/LonCfbroREBgbyuApJUIjbbXKtTOv/YTzw4LxTbxLVmqR0YzVuI4fjnq7GxJbmLRXt/FDFCxdZ/q8CqR1tIM0jgqAMwcVGhUswHhAl6Y2r5hzUPowauVJ3TKc7Rv9OJxG6bklZY8CNqeyU/3subvGuxnAhBwBm/YqTf9N2Lmvbl7cBFt6T5OHHjKuIT6vWJR2nSrcfJIdQbhu1cxwE5PGxMuzwAcNdoFwlthmX1q1vuH4RKV2+V20wn3szGnTbEd2nOV2Q/VyaBZ3/9vMAXMiRH9HwKE0m5LYnDFhOVwpuU8eZR7qqkagEg8XJquLwIpnEOz51ZCr99OCehtlG6z/F/qI4QrC/ncrdHZY+MCACFB42SKDtz/8RyZlJudTWD+KI3QuoUlKjQAAu34aT6Bkol6b0JJsIbWFRNum642Z2G62OACvrQiq/d0XadI4+3nhHhRDIiMmFjfHb8K2Zqj5JNcp8JJ9POCJy33kmbDlIRP4iYTxjXgH6gBdhIBk9oM6Yjtvx8bncv4wcSjGTuD7TsYOVsWHfF0UtGdANHRrSb4gTMID6aIpxkEYml/Iu1cweTJqwhKXyI2FS7aF4/sWYIVcaUzP/78OsfEcmdSphTzUzSJFtHQit8MmdG0Gpp1Rb+do3OlHvz9ANaKDnvDxc/c3Po57pLO+DnaIBzprHvVUkzrByXKwEKQUxJXAEWDxP9cQ38Ph9Hzdtopqamov/TcdXdnHcrzaqc1qwUET9m2IoCenutZLyfx0tvmGJiGylSu7vmTkGG2F+HraVOvoSdo8eeFu1EFAP+b+KLQNZ/RtEsSyuYgUJ/MeziQ4mFST2YrDmjtTzvh2ZaFTSMuKstRqEnddAOLeJcf1m26L82kh3tl2ckwkHeKLReUMxH+FFIIpNe8aFfyzEFsVZ89ZG98stCr5iNmxzPYm+71STyPMaFZ2LokJOlOZHzJXuyqQR9KHXzmUZFCAo7szob4vatukPs4iowiGBcqVhLubUlU8BT4oqkaYpNmNvMIEnwzDhAedLMm8iKbPJR2Fk2G3GBhlQbSojqd2NFyV+OAG6bMHcePXPYR1jCl1SXDZTg01L8XfeRTwwDSrSMBBMiq8UbGkPawZcWYC5GN5A9HEhH+0cxEdCy9lBufc
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 6fe74b40-3715-4ee5-38aa-08d8eef40c54
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3021.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 24 Mar 2021 18:38:42.0861
 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 E1sItQcec1iiMDgZlSY0Xt4Kb3bkBbIgeIOEZhSVVsi2nDGy2pcN48LlGoszy5XS5Dgl8TvPRH1g9Sm9HCPtkM/zMoblmwItiJ1LsIsAbJQ=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SA2PR10MB4665
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9933
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 adultscore=0
 malwarescore=0 spamscore=0
 mlxscore=0 phishscore=0 suspectscore=0 mlxlogscore=999 bulkscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2009150000
 definitions=main-2103240133
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9933
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 adultscore=0
 malwarescore=0 mlxscore=0
 priorityscore=1501 bulkscore=0 impostorscore=0 lowpriorityscore=0
 phishscore=0 mlxlogscore=999 suspectscore=0 clxscore=1015 spamscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2009150000
 definitions=main-2103240133
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

These #defines will be used by nested.c in the next patch. So move these to
svm.h.

Signed-off-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
---
 arch/x86/kvm/svm/svm.c | 3 ---
 arch/x86/kvm/svm/svm.h | 3 +++
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/x86/kvm/svm/svm.c b/arch/x86/kvm/svm/svm.c
index 58a45bb139f8..a3d8699e70d6 100644
--- a/arch/x86/kvm/svm/svm.c
+++ b/arch/x86/kvm/svm/svm.c
@@ -56,9 +56,6 @@ static const struct x86_cpu_id svm_cpu_id[] = {
 MODULE_DEVICE_TABLE(x86cpu, svm_cpu_id);
 #endif
 
-#define IOPM_ALLOC_ORDER 2
-#define MSRPM_ALLOC_ORDER 1
-
 #define SEG_TYPE_LDT 2
 #define SEG_TYPE_BUSY_TSS16 3
 
diff --git a/arch/x86/kvm/svm/svm.h b/arch/x86/kvm/svm/svm.h
index 39e071fdab0c..ae0a629b3ec6 100644
--- a/arch/x86/kvm/svm/svm.h
+++ b/arch/x86/kvm/svm/svm.h
@@ -28,6 +28,9 @@ static const u32 host_save_user_msrs[] = {
 };
 #define NR_HOST_SAVE_USER_MSRS ARRAY_SIZE(host_save_user_msrs)
 
+#define IOPM_ALLOC_ORDER 2
+#define MSRPM_ALLOC_ORDER 1
+
 #define MAX_DIRECT_ACCESS_MSRS	18
 #define MSRPM_OFFSETS	16
 extern u32 msrpm_offsets[MSRPM_OFFSETS] __read_mostly;

From patchwork Wed Mar 24 17:50:03 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Krish Sadhukhan <krish.sadhukhan@oracle.com>
X-Patchwork-Id: 12162009
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,MSGID_FROM_MTA_HEADER,SPF_HELO_NONE,
	SPF_PASS,USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id A556EC433F1
	for <kvm@archiver.kernel.org>; Wed, 24 Mar 2021 18:39:33 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 7A4CD61A26
	for <kvm@archiver.kernel.org>; Wed, 24 Mar 2021 18:39:33 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S237724AbhCXSjB (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Wed, 24 Mar 2021 14:39:01 -0400
Received: from aserp2120.oracle.com ([141.146.126.78]:49174 "EHLO
        aserp2120.oracle.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S237686AbhCXSir (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 24 Mar 2021 14:38:47 -0400
Received: from pps.filterd (aserp2120.oracle.com [127.0.0.1])
        by aserp2120.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 12OIPqO7047526;
        Wed, 24 Mar 2021 18:38:45 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : in-reply-to : references :
 content-transfer-encoding : content-type : mime-version;
 s=corp-2020-01-29; bh=dyX7Z/sm1wFUhxRuIDoHhXtN3HGD+gw+OWnMFUMsc5Y=;
 b=REjmLRh81rflq3eZGVOGpFrJK5z/wvt8Nq/LkNnxcy0wcZB/Te6wamnT59Ah7nWhpbNu
 xzAEQh04bQWUK35Zv6iIAS2oeH/Nzh6qOQRkLS5K1PCkMFZV7vEK2MlU5lSVslWjFJ4r
 hcFn9hu7Cc23NpvE5awOX52xLg5N+4SvqArBJ09ITwrEruVveEf4ElRPfmn8W5xkvkLF
 EjW+xrk7o/160sbdJ6FpptVByKp7EzSWS24v7MQ4YtbD0HUO+fcZaN/RnAXUgy8wALRq
 tMxt6CfdlB8A4oDChnwnJdL7KxJgovpkkwz9P+Ixi9wf6SnZsASNI+VDlwXQO0KQhkVp 7Q==
Received: from userp3020.oracle.com (userp3020.oracle.com [156.151.31.79])
        by aserp2120.oracle.com with ESMTP id 37d90mktnw-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Wed, 24 Mar 2021 18:38:44 +0000
Received: from pps.filterd (userp3020.oracle.com [127.0.0.1])
        by userp3020.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 12OIQFYK167607;
        Wed, 24 Mar 2021 18:38:44 GMT
Received: from nam12-mw2-obe.outbound.protection.outlook.com
 (mail-mw2nam12lp2040.outbound.protection.outlook.com [104.47.66.40])
        by userp3020.oracle.com with ESMTP id 37dtttpj5x-3
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Wed, 24 Mar 2021 18:38:44 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=ZK4Z71fg+ymP0vNmEUTTujZ6s/Mb6p8NN3XfRC4OSO8B/ka5BFW7ggs+DIE4PM9GVpn9AbEXVp09JsinI5Hb9+PKgsbbMgsiCyiAJJDvIMSQf5pksvza1MwTV+nIeWupLrxIINHIFyrJm7Gm5sYwRkmsEeeDomELlAK1GNkxlCvxwGe/3HUfyQjpSfLlydncKY2rbuPdjxJh0ljuinzt9q3/0wiv9kAyXKCJn5lYHBWhgD7FzGT6qBiAHzmpfLcwNagW0FrAUmmjJt79T0LCdtQOqjAsTLPp0zCjbXiPGPsCTmmPKk7HMxcVWq1Qig8HCrDaVc4vmiTR7PxmGne5qA==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=dyX7Z/sm1wFUhxRuIDoHhXtN3HGD+gw+OWnMFUMsc5Y=;
 b=RL3pYnH9JqqVj1BgawxGL80aVCpndOOXEYt2hNLj4GMRBjUHrTpzoso5tKKunz15NZa01So3bezknp81OL3NKU06K/fl4yQhBVYmxP+VqwQT09oXy300f2bxuXQ9gXhp6iYwAbv2cXpMJ5hJVVnhSP46B50tJ1BIdvCXl9qWISuhMWKrxrZDM4pDzfzQxdq+evYArAyTjEHlB1JNUZEBfv5Kb63UQc1q/1xWfgSculbehOY9PIJ1NmXg1K7QUMVbvMXWtxZXvFpTHOgZJWGvfeTLH4gY1mFvDANNGeJv71gIEiwRLVKCOd6nTT2atqlH/nDNmRgi+X661lJpaAH36A==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=dyX7Z/sm1wFUhxRuIDoHhXtN3HGD+gw+OWnMFUMsc5Y=;
 b=EaooH/0e04iKJTGsgnF/+pu9fKXbcCTKtKyeV38/iU/Sc9hxAkNRxGVVQYQ51w8iQnhN0V8DmsU6876IC3FYsFr6kjrn8rFEtZtKXqdSdzVQr7RL3e7wY1RdkBw6zTgwLT72p5LoHSmQ8l98TXij5TmnLz8IwAC+4gqfeuo3QbQ=
Authentication-Results: vger.kernel.org; dkim=none (message not signed)
 header.d=none;vger.kernel.org; dmarc=none action=none header.from=oracle.com;
Received: from SN6PR10MB3021.namprd10.prod.outlook.com (2603:10b6:805:cc::19)
 by SA2PR10MB4665.namprd10.prod.outlook.com (2603:10b6:806:fb::17) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3955.18; Wed, 24 Mar
 2021 18:38:43 +0000
Received: from SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::1871:3741:cc17:bcf7]) by SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::1871:3741:cc17:bcf7%7]) with mapi id 15.20.3955.025; Wed, 24 Mar 2021
 18:38:43 +0000
From: Krish Sadhukhan <krish.sadhukhan@oracle.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, jmattson@google.com, seanjc@google.com
Subject: [PATCH 2/5 v4] KVM: nSVM: Check addresses of MSR and IO permission
 maps
Date: Wed, 24 Mar 2021 13:50:03 -0400
Message-Id: <20210324175006.75054-3-krish.sadhukhan@oracle.com>
X-Mailer: git-send-email 2.25.4
In-Reply-To: <20210324175006.75054-1-krish.sadhukhan@oracle.com>
References: <20210324175006.75054-1-krish.sadhukhan@oracle.com>
X-Originating-IP: [138.3.201.29]
X-ClientProxiedBy: BYAPR05CA0009.namprd05.prod.outlook.com
 (2603:10b6:a03:c0::22) To SN6PR10MB3021.namprd10.prod.outlook.com
 (2603:10b6:805:cc::19)
MIME-Version: 1.0
X-MS-Exchange-MessageSentRepresentingType: 1
Received: from ban25x6uut29.us.oracle.com (138.3.201.29) by
 BYAPR05CA0009.namprd05.prod.outlook.com (2603:10b6:a03:c0::22) with Microsoft
 SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3977.10 via Frontend
 Transport; Wed, 24 Mar 2021 18:38:42 +0000
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 82019940-9176-4428-da8d-08d8eef40cf7
X-MS-TrafficTypeDiagnostic: SA2PR10MB4665:
X-Microsoft-Antispam-PRVS: 
 <SA2PR10MB4665D70C7D3A04FC96B52BA681639@SA2PR10MB4665.namprd10.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:5236;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 M2giGfAJ8SyTIxfCNPizuUVz0082XszKn7PYTPF3iF9lhv5d82YK+inJi/cnZv92R4ZTYfY7SMkcBYlAZUbwFx1TmSdbkRM7TlmODgk3S+7Nx3v0Gb5TiF/r4JQKVvIkitOhuUHKe2kvGIBvMMZBaTkGRiMpODYwpwBlhmqQDLdu3taHwSiJLCKqGEZyJDXpe5Z3+SLNyqRqgSOGruitAo7Q5awDhUiFZqAdr8DbCRCrcsesJ71WCxci68VrMZtRRDnbCJ8qnIaUd3IYhwAu/q9sflpXcpmJATZYVeJYAoA3PRa9sFafA66l/k8M+nNlY95/ZCFCxjyTU8K+2XA+Tko+52+qbGh0tdhATQbRKJjOJ68DNaoG5HtnVF1EcepxxJ6QnbvQ4BvUaqb6evoW/8/xxsd7VB77EjYfzbU/YsOycuAZaTdTrxMUduJx2avDQcdIcN0AB7ikoLgVBPcUdzqfsq7oeJ2UBFpy3FaRek827qi2U1qN6SDb0XpguFYjRV221wQaEGDkB3+exWnsg4bcqP5ibREIaxjGHi1w4niN9AO73dueKG3Kp07Uk5XJ1g+IaZTNf1ijvFCHdIxp8qQy3wqkAeARIkGM7IZOsWDrV472qz23l9ij/p3xXXvaPsp8mrvaTLGWBNI0412nrA==
X-Forefront-Antispam-Report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3021.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(376002)(346002)(396003)(366004)(39860400002)(136003)(66946007)(6486002)(36756003)(16526019)(186003)(478600001)(6666004)(956004)(1076003)(6916009)(2616005)(8676002)(38100700001)(4326008)(83380400001)(44832011)(316002)(66476007)(66556008)(5660300002)(2906002)(7696005)(52116002)(8936002)(86362001)(26005);DIR:OUT;SFP:1101;
X-MS-Exchange-AntiSpam-MessageData: 
 AE/VZaI36DEbvo3eGK8J1dJ4xn09n02Xpc893QIinEkEkr8ESZBzSsLEonuuaQUIgsHx8EEvjOqycsh/QIbmCO8x3VBRWO+4Zk48m0nRD5CQfcdthjSvNijdL2CMgcmHSWqunQlAb/wDsDPK7N87IMub/59C/F+GrYNGMVgQWvskRvGUKiWmR9ly3u3vzSLnMu7uHrK1kHoAKSZCktU4ryNLPH18aMxWqGiZqYC469bKjQbie7eFb9xpFnAoTJoRYuYRaudE/UA8kUnZBzaC75h6u2H2+QDMcvINaFNZmdMBst9vRI1D9yIk9E3ayW4WiftOAfpjZULUFUu+UcxQL/IYAyWbzHVD5fn14EjheG3fMdeAMgpRI0UXjbS66nzYw5EUBkh5JH1vDFe9vit7sBdhdUxsUf/54BGpvkau+USesSBFo7gO9HWJr5k9TJRMxvEHY19v9zC4KVb/x85XcCq+5s2rD1hTYOwSVubE7JHAsjHVyLp5wZrfZ4qblR7UFks4ECjWJFx73o+SjFoYBjFFvpeyO2Z7upDuggEL0xRF/qBixr9E4e5WeepUwNua/pfDL4uipt2lCjZCgHTB1hUq49mx0X83eCTOamT4RXdCntzbPYAYF0Yyo2r2dRp9RITBltwom1N7YSanwZCdw7ONDq/oqsoPbfx+VaDk1zagXMQKnAd2LnYHRg7mEFPU5Wdzq8uuPCmfgkit8//sHGMmmb4rqrDrua+1NfzVjRUtGPxFdKu9cBnzKiiY9gKqzWMs6s8P11rf4YRhliZOmzVpEKlwWah9yMsXzzGGdwuV5iTrL4Ge3qLE3ALXoL5A5ICPYRLkXoYGVYA6NFAXPBctYnqNWdkqpwudZI5bilZvBd8Dv/8cTJ2dN/fJDTSfzQIrnfNfy/FiEp7eMYjdC5pYtAESc0DfxAcxvfjL7YbhliD9/XOLsKblWBYr2gYvA3OOoRbXwPWIup+Supw2v78BPMXswx53rrB6ZNUPRS12o9gh4tkaKY7i7F9cDgf3CB0pASQnAQkRKPUWs4xhAqvidLZGenVtljI46kSWpIc3F9mYHuGp8+GnSgLRCsvVK/ulSBM+2qiin3klS6cIFO9idq8OjFN+CXCvKCpipp1mx7bzNlX3fS29hjaKAwn4/GWnJ/LUGBslUAlI+YkTBUlkcbipnXiKCv5HmRZ5jx9UrbTARdXrbAn9RSh61eY+CJpfOc/B4yy6ebHplN91kx+lNHaUTBSG9hGWDWhBxFwsZSduIjZgKH8sCyQhIPWnVorsVQNVPtfF3WAt0ip8z5d0teJV5feDpvBV+D8oEADayfv+3mH0hhRZNRq0t37f
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 82019940-9176-4428-da8d-08d8eef40cf7
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3021.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 24 Mar 2021 18:38:43.1615
 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 Gye4mnQCOMxar6lGEBQy2tgoKdo/IBoqa6WPF2gOmrpq6scM7e9jWEeunHmVRWVl3dQ3JQRoUYIGvu90YqHXDpzsb3ku8ECRXgtezmG4/5s=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SA2PR10MB4665
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9933
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 adultscore=0
 malwarescore=0 spamscore=0
 mlxscore=0 phishscore=0 suspectscore=0 mlxlogscore=999 bulkscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2009150000
 definitions=main-2103240133
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9933
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 adultscore=0
 malwarescore=0 mlxscore=0
 priorityscore=1501 bulkscore=0 impostorscore=0 lowpriorityscore=0
 phishscore=0 mlxlogscore=999 suspectscore=0 clxscore=1015 spamscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2009150000
 definitions=main-2103240133
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

According to section "Canonicalization and Consistency Checks" in APM vol 2,
the following guest state is illegal:

    "The MSR or IOIO intercept tables extend to a physical address that
     is greater than or equal to the maximum supported physical address."

Suggested-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
---
 arch/x86/kvm/svm/nested.c | 28 +++++++++++++++++++++-------
 1 file changed, 21 insertions(+), 7 deletions(-)

diff --git a/arch/x86/kvm/svm/nested.c b/arch/x86/kvm/svm/nested.c
index 35891d9a1099..b08d1c595736 100644
--- a/arch/x86/kvm/svm/nested.c
+++ b/arch/x86/kvm/svm/nested.c
@@ -231,7 +231,15 @@ static bool svm_get_nested_state_pages(struct kvm_vcpu *vcpu)
 	return true;
 }
 
-static bool nested_vmcb_check_controls(struct vmcb_control_area *control)
+static bool nested_svm_check_bitmap_pa(struct kvm_vcpu *vcpu, u64 pa,
+				       u8 order)
+{
+	u64 last_pa = PAGE_ALIGN(pa) + (PAGE_SIZE << order) - 1;
+	return last_pa > pa && !(last_pa >> cpuid_maxphyaddr(vcpu));
+}
+
+static bool nested_vmcb_check_controls(struct kvm_vcpu *vcpu,
+				       struct vmcb_control_area *control)
 {
 	if ((vmcb_is_intercept(control, INTERCEPT_VMRUN)) == 0)
 		return false;
@@ -243,12 +251,18 @@ static bool nested_vmcb_check_controls(struct vmcb_control_area *control)
 	    !npt_enabled)
 		return false;
 
+	if (!nested_svm_check_bitmap_pa(vcpu, control->msrpm_base_pa,
+	    MSRPM_ALLOC_ORDER))
+		return false;
+	if (!nested_svm_check_bitmap_pa(vcpu, control->iopm_base_pa,
+	    IOPM_ALLOC_ORDER))
+		return false;
+
 	return true;
 }
 
-static bool nested_vmcb_checks(struct vcpu_svm *svm, struct vmcb *vmcb12)
+static bool nested_vmcb_checks(struct kvm_vcpu *vcpu, struct vmcb *vmcb12)
 {
-	struct kvm_vcpu *vcpu = &svm->vcpu;
 	bool vmcb12_lma;
 
 	if ((vmcb12->save.efer & EFER_SVME) == 0)
@@ -268,10 +282,10 @@ static bool nested_vmcb_checks(struct vcpu_svm *svm, struct vmcb *vmcb12)
 		    kvm_vcpu_is_illegal_gpa(vcpu, vmcb12->save.cr3))
 			return false;
 	}
-	if (!kvm_is_valid_cr4(&svm->vcpu, vmcb12->save.cr4))
+	if (!kvm_is_valid_cr4(vcpu, vmcb12->save.cr4))
 		return false;
 
-	return nested_vmcb_check_controls(&vmcb12->control);
+	return nested_vmcb_check_controls(vcpu, &vmcb12->control);
 }
 
 static void load_nested_vmcb_control(struct vcpu_svm *svm,
@@ -515,7 +529,7 @@ int nested_svm_vmrun(struct vcpu_svm *svm)
 	if (WARN_ON_ONCE(!svm->nested.initialized))
 		return -EINVAL;
 
-	if (!nested_vmcb_checks(svm, vmcb12)) {
+	if (!nested_vmcb_checks(&svm->vcpu, vmcb12)) {
 		vmcb12->control.exit_code    = SVM_EXIT_ERR;
 		vmcb12->control.exit_code_hi = 0;
 		vmcb12->control.exit_info_1  = 0;
@@ -1191,7 +1205,7 @@ static int svm_set_nested_state(struct kvm_vcpu *vcpu,
 		goto out_free;
 
 	ret = -EINVAL;
-	if (!nested_vmcb_check_controls(ctl))
+	if (!nested_vmcb_check_controls(vcpu, ctl))
 		goto out_free;
 
 	/*

From patchwork Wed Mar 24 17:50:04 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Krish Sadhukhan <krish.sadhukhan@oracle.com>
X-Patchwork-Id: 12162011
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,MSGID_FROM_MTA_HEADER,SPF_HELO_NONE,
	SPF_PASS,USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 3A23FC43445
	for <kvm@archiver.kernel.org>; Wed, 24 Mar 2021 18:39:34 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 238AE61A25
	for <kvm@archiver.kernel.org>; Wed, 24 Mar 2021 18:39:34 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S237750AbhCXSjF (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Wed, 24 Mar 2021 14:39:05 -0400
Received: from aserp2130.oracle.com ([141.146.126.79]:50964 "EHLO
        aserp2130.oracle.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S237689AbhCXSiu (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 24 Mar 2021 14:38:50 -0400
Received: from pps.filterd (aserp2130.oracle.com [127.0.0.1])
        by aserp2130.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 12OIOR4c117165;
        Wed, 24 Mar 2021 18:38:47 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : in-reply-to : references :
 content-transfer-encoding : content-type : mime-version;
 s=corp-2020-01-29; bh=0BmbeEmUbLeyXPR4iuqm2o4WOrw+pqilh+OZEuhBVIo=;
 b=gPpOfuCAyiASTf92K03dDad0610K0SelM+scxsm+sR6DVwWsEYqbOJ/r3siXv6A5xPRP
 bnpOWQLAIfrtunZzfnYZCJuq31Gr4qfOz1xMrhWDlxSwKoXPaMWhSAiu8cvmBsbMT7rL
 wDsxvANeHiXqH+jYAQjQCEmWOIwcSvD9isHVuRyPKgdhkG/0yV1Eh9N874A2U+f+GdgB
 ikSqxa4erjHSqLMEncPh4YHc7SLJa7AOKNaIKbCzCcRLIymXfGTV60LWpONDoV1+sWex
 NF9d/fkJXAMN8uvSeRAxWyl9iKeRfjz9EmprDjYSE2Jh3DMCgDy7Y0sU9e/8KNKWKxt9 Pg==
Received: from userp3020.oracle.com (userp3020.oracle.com [156.151.31.79])
        by aserp2130.oracle.com with ESMTP id 37d6jbkxx3-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Wed, 24 Mar 2021 18:38:47 +0000
Received: from pps.filterd (userp3020.oracle.com [127.0.0.1])
        by userp3020.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 12OIQFPY167563;
        Wed, 24 Mar 2021 18:38:46 GMT
Received: from nam04-dm6-obe.outbound.protection.outlook.com
 (mail-dm6nam08lp2046.outbound.protection.outlook.com [104.47.73.46])
        by userp3020.oracle.com with ESMTP id 37dtttpj74-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Wed, 24 Mar 2021 18:38:46 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=dLozW9Yn/QFenYo0kwVShVBqhVKWiSex323k9Zgz7/uClQtmMfkxuFnh5HSxbBUQOXLLIVsi8cWd6Kxu0t8bpcmuZVpFmANYU5kzmXEUU9Lsuy9Z0MFOptXFpxRxjTKbn6dtRzkkDPM1szTnZbVzqVEjG09o0TO6sqTHjsSOQk0uBZIEY5+/Juk73Twc22vADl2kPhovBx7yah/kOBlywTJGzeB/PutEgWJjLT1T77ZWgOFTNovLyo1+ClHqWS0N4eJtkToDsSpQHemDRX4iFFA1Lhk+BMHAX8Xd1dmACC9kilFzjqTATYd+QMXfhSYB5bXK69U1vjNKEAK/kp2aEw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=0BmbeEmUbLeyXPR4iuqm2o4WOrw+pqilh+OZEuhBVIo=;
 b=mvuvDGYbY7WFoPwQ+ve28+Fuyc8GRQIJ0Gp/IoUcdlDY897R12ZAaLfemHPmSMpXO279NULl/jl1ymwxSs6GfaQkKmay7if3tpxPajQUEnJWFHVp4Od7U3xW/AkyDJdpsVoZlxfvpr02kK/NUzZHpZMskn789Eo8HrbEx4c9/id5hhsZfQ9xyZwrjyL86Y8Mx5MNt37TsDOzsrqT8oQn4JgBksU1RQlivEo6xebzIbGUHAqhN/kSlnLOqwKzaN47/gZb9KbSPcJd7E6xaVNZEaVbWrApMMxXRQG0VdJgZw+Tvh0GI2Zo2XE/xDx/mn+6Rj1Y7xeLbWHjfvxtxNNz1A==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=0BmbeEmUbLeyXPR4iuqm2o4WOrw+pqilh+OZEuhBVIo=;
 b=NSA0bmiJM8ifUATOr23tC7/qGHUZVIFllByvvbyYVDChj/Gu3/NGiggcF2/Xdv3dxndTH+AjNZbdcVN3lTzHspR0cykMdzlzUY0kNZ3sPQ5xVme852NlnfjvMbZLXjcJgrJBBitY33txrnYGDQ1DynMPBq0//pT9CO1RkgfbL3M=
Authentication-Results: vger.kernel.org; dkim=none (message not signed)
 header.d=none;vger.kernel.org; dmarc=none action=none header.from=oracle.com;
Received: from SN6PR10MB3021.namprd10.prod.outlook.com (2603:10b6:805:cc::19)
 by SA2PR10MB4425.namprd10.prod.outlook.com (2603:10b6:806:11b::9) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3977.24; Wed, 24 Mar
 2021 18:38:44 +0000
Received: from SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::1871:3741:cc17:bcf7]) by SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::1871:3741:cc17:bcf7%7]) with mapi id 15.20.3955.025; Wed, 24 Mar 2021
 18:38:44 +0000
From: Krish Sadhukhan <krish.sadhukhan@oracle.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, jmattson@google.com, seanjc@google.com
Subject: [PATCH 3/5 v4] KVM: nSVM: Cleanup in nested_svm_vmrun()
Date: Wed, 24 Mar 2021 13:50:04 -0400
Message-Id: <20210324175006.75054-4-krish.sadhukhan@oracle.com>
X-Mailer: git-send-email 2.25.4
In-Reply-To: <20210324175006.75054-1-krish.sadhukhan@oracle.com>
References: <20210324175006.75054-1-krish.sadhukhan@oracle.com>
X-Originating-IP: [138.3.201.29]
X-ClientProxiedBy: BYAPR05CA0009.namprd05.prod.outlook.com
 (2603:10b6:a03:c0::22) To SN6PR10MB3021.namprd10.prod.outlook.com
 (2603:10b6:805:cc::19)
MIME-Version: 1.0
X-MS-Exchange-MessageSentRepresentingType: 1
Received: from ban25x6uut29.us.oracle.com (138.3.201.29) by
 BYAPR05CA0009.namprd05.prod.outlook.com (2603:10b6:a03:c0::22) with Microsoft
 SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3977.10 via Frontend
 Transport; Wed, 24 Mar 2021 18:38:43 +0000
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: c81cf52e-1e82-433b-e421-08d8eef40da1
X-MS-TrafficTypeDiagnostic: SA2PR10MB4425:
X-Microsoft-Antispam-PRVS: 
 <SA2PR10MB442537F51ED1A7E7C4B8714181639@SA2PR10MB4425.namprd10.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:4714;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 6dbT6fr++3p/K+lv+0L7VaSWi29bgrtnmOtJU6p90D/TK6KVNIjq73UbwjJz/Dr8ex5ZERF4YNe7dt3VhH3ozA6wl6xxoOyVVNljh45DA6ijmFM8hlP/sONSeAcFz1rSdJ8hcNIo54nCaEa6KGgAFJ9lQoLJRMSO2AZ6ChuJlJmiTkTPjC3pdGGXnfmme9KvPMUQhgcHosvNLL2XVufNfVAeWMwNhP/o9LIJRmpN8hnPf+clhBLb1x3aYQM8ZSpGMQdpWapRG4KlH+WMzqqxQgrozIkMIuGVy2ujXBVZ+5ysc6UbiSnOX1Vb8cfl4Fh/LsPWZ5v/cZS/jefwgtTitfqz1b7+JQbKrvDjsSb61Oqd1RTEatOA0tU1Pa/bACB11vRQXl3eBj/S4bbCbZB8VcKii3F4zQfyV7N7bcfRA6dwpJBVDTSI0HBWx6+1aXMZbqigkcVSensrRscbVT4MI4iFLoNUDjNzM1dq4qcsX37A07la/BbCCxuH3UUYtAEqhuQc3x30trjUowIOH6EPiXu3gIr13C0XL9F5iutBRQgICW6qS5WCohmrFpqIU9xPskho39Xrwv6XGAFz8Gii5/mEdkYJEXn/JkNTzu8uN8Fl7/nOQ1iOhp2gNIt/O3yRWQSCU4PSnsBg12H9nRE8pg==
X-Forefront-Antispam-Report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3021.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(136003)(346002)(376002)(366004)(396003)(39860400002)(38100700001)(52116002)(8936002)(7696005)(6666004)(36756003)(1076003)(5660300002)(6486002)(66946007)(66476007)(8676002)(66556008)(6916009)(2906002)(86362001)(26005)(83380400001)(956004)(186003)(478600001)(44832011)(16526019)(4326008)(2616005)(316002);DIR:OUT;SFP:1101;
X-MS-Exchange-AntiSpam-MessageData: 
 Ww4VvLBSXLIt/4t+B05y08IHOYrX8IJdqCF8whlrdyxLZaV6eTNdPzj5MQ0RQrS/4KpGjnhXAIWlNyek4ULWcpUN5N6XBhWe00HYgiIM/ruOQl5bZi8MG5039maHfVeTFDsIx8uSzF5sr0+WpkFAzYp8ojA0LmR2lJp/zcPH+jByFG7GWARji6M+kAZLKP3Y+4TM4lF0/AYU9OLnYD5SS6aUcKIcZlrlOQpqnhmszgkJPXRZP1BWUgfxMsavLxtEceROeVDUVXHMJnRdiC4uvQVYOxUEnzfq6p8oMADPvxgqNZAYm7XngVo2rBceX/ej+vWeGHriLXbWIh4dF6wDIkOSoJpcmnfSuWlw5Q+PhIi+KRzAAcwt2CuT2yHsJk8/kNQ8JHToMjyT4wbmZqj3/zwc1xWXrsDAj8vss6kDqChyCKiQMZ5uVN6jeFl0VJ6n85Hl7pExD6kXpL7OOkVeME1TZrqEvTM6W7u9nOI+0n+Wk9r0U7H1pVly9q7XXsO1DZUKoKgH5FxDRxg54Z9f2eW+yVYJVvBowisjIu0rV3Ra5+JgTusZv/eEaHy/8B9259LuoCkG5clx14iLmegePJ7qseQKyxwahVey+shr77Pmrl2c3Gqzmmk7x9zzcubRhekEsKIM6eqXHia/ZBmGyhJLcPKi0fdkNbEwf2v9cavqUqXDZJ8pSreb8+QJaWCSvr05980ylZOulU6BEg2felCCr3HFKxdAMv78b+rqsCuSg6kYDgiY2WnXFCbiKAgHPcgEl+ZL0+DeBn+nXbbjX6/VGqamk8RK0xLAcb7PNpZ5JF1tu4IifXj66JHMcN6AZRks+uZHaVMnJhMTWHfiaCFZGIql/rayVn+OrZvW+acMo02cXeZjwDWrdkH/tV7xsy2vZr7/2iNc9tT66v92iZZq0NFysuqpxQyHSrcEb59fcG9lcPagp8gtna/iaR7lKEZJr5EI4+MsLKUy/S+7hFP88BDeiJcjTYHRJCfYPQ42huITOUGTDO39nRqVVQP8c4VjypZ9iO81OQBMkoFInu3OSzAvSKUSq13bFFQZ2fKRUK+IhENuLpJPcQqfUI2lOGxbbPdh4GESoQjFnf6a7JqJPzGvrLH81Njn1J44/F1Ai+Dp/jn055TaDELg6l5U4Kkd0WsvAL7fBVhcKp0INYhWMoJ8fVmu/5kcKaLs/LwQYoDxLMSG75FLDXBazq//cXWF2dSv5DuDRb+3Dgd5NNPGd7woB9H9Kq4OMZ8ggx/gMEhkmgAIMCdBFctC78btgyIZHoa4/m7D95cS/5Xu2F/aogWbaQmSt/Et6wqilES63IqvXlCiZNHWYuiQiOrN
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 c81cf52e-1e82-433b-e421-08d8eef40da1
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3021.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 24 Mar 2021 18:38:44.3887
 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 xTecIhU5ojDZ0ePQFQYTCxmstt4I08e6rXsqGJnBnBZqUpkWXhf295/VMionHfBztrFlSWR1SgpOuqjj8c0zeodSvFNgI4y598kiguL6qGA=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SA2PR10MB4425
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9933
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 adultscore=0
 malwarescore=0 spamscore=0
 mlxscore=0 phishscore=0 suspectscore=0 mlxlogscore=999 bulkscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2009150000
 definitions=main-2103240133
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9933
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 bulkscore=0
 phishscore=0
 lowpriorityscore=0 suspectscore=0 clxscore=1015 priorityscore=1501
 spamscore=0 adultscore=0 impostorscore=0 mlxlogscore=999 mlxscore=0
 malwarescore=0 classifier=spam adjust=0 reason=mlx scancount=1
 engine=8.12.0-2009150000 definitions=main-2103240133
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Use local variables to derefence svm->vcpu and svm->vmcb as they make the
code tidier.

Suggested-by: Sean Christopherson <seanjc@google.com>
Signed-off-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
---
 arch/x86/kvm/svm/nested.c | 33 +++++++++++++++++----------------
 1 file changed, 17 insertions(+), 16 deletions(-)

diff --git a/arch/x86/kvm/svm/nested.c b/arch/x86/kvm/svm/nested.c
index b08d1c595736..a02a4e01e308 100644
--- a/arch/x86/kvm/svm/nested.c
+++ b/arch/x86/kvm/svm/nested.c
@@ -503,33 +503,34 @@ int nested_svm_vmrun(struct vcpu_svm *svm)
 {
 	int ret;
 	struct vmcb *vmcb12;
+	struct kvm_vcpu *vcpu = &svm->vcpu;
 	struct vmcb *hsave = svm->nested.hsave;
 	struct vmcb *vmcb = svm->vmcb;
 	struct kvm_host_map map;
 	u64 vmcb12_gpa;
 
-	if (is_smm(&svm->vcpu)) {
-		kvm_queue_exception(&svm->vcpu, UD_VECTOR);
+	if (is_smm(vcpu)) {
+		kvm_queue_exception(vcpu, UD_VECTOR);
 		return 1;
 	}
 
-	vmcb12_gpa = svm->vmcb->save.rax;
-	ret = kvm_vcpu_map(&svm->vcpu, gpa_to_gfn(vmcb12_gpa), &map);
+	vmcb12_gpa = vmcb->save.rax;
+	ret = kvm_vcpu_map(vcpu, gpa_to_gfn(vmcb12_gpa), &map);
 	if (ret == -EINVAL) {
-		kvm_inject_gp(&svm->vcpu, 0);
+		kvm_inject_gp(vcpu, 0);
 		return 1;
 	} else if (ret) {
-		return kvm_skip_emulated_instruction(&svm->vcpu);
+		return kvm_skip_emulated_instruction(vcpu);
 	}
 
-	ret = kvm_skip_emulated_instruction(&svm->vcpu);
+	ret = kvm_skip_emulated_instruction(vcpu);
 
 	vmcb12 = map.hva;
 
 	if (WARN_ON_ONCE(!svm->nested.initialized))
 		return -EINVAL;
 
-	if (!nested_vmcb_checks(&svm->vcpu, vmcb12)) {
+	if (!nested_vmcb_checks(vcpu, vmcb12)) {
 		vmcb12->control.exit_code    = SVM_EXIT_ERR;
 		vmcb12->control.exit_code_hi = 0;
 		vmcb12->control.exit_info_1  = 0;
@@ -539,8 +540,8 @@ int nested_svm_vmrun(struct vcpu_svm *svm)
 
 
 	/* Clear internal status */
-	kvm_clear_exception_queue(&svm->vcpu);
-	kvm_clear_interrupt_queue(&svm->vcpu);
+	kvm_clear_exception_queue(vcpu);
+	kvm_clear_interrupt_queue(vcpu);
 
 	/*
 	 * Save the old vmcb, so we don't need to pick what we save, but can
@@ -552,17 +553,17 @@ int nested_svm_vmrun(struct vcpu_svm *svm)
 	hsave->save.ds     = vmcb->save.ds;
 	hsave->save.gdtr   = vmcb->save.gdtr;
 	hsave->save.idtr   = vmcb->save.idtr;
-	hsave->save.efer   = svm->vcpu.arch.efer;
-	hsave->save.cr0    = kvm_read_cr0(&svm->vcpu);
+	hsave->save.efer   = vcpu->arch.efer;
+	hsave->save.cr0    = kvm_read_cr0(vcpu);
 	hsave->save.cr4    = svm->vcpu.arch.cr4;
-	hsave->save.rflags = kvm_get_rflags(&svm->vcpu);
-	hsave->save.rip    = kvm_rip_read(&svm->vcpu);
+	hsave->save.rflags = kvm_get_rflags(vcpu);
+	hsave->save.rip    = kvm_rip_read(vcpu);
 	hsave->save.rsp    = vmcb->save.rsp;
 	hsave->save.rax    = vmcb->save.rax;
 	if (npt_enabled)
 		hsave->save.cr3    = vmcb->save.cr3;
 	else
-		hsave->save.cr3    = kvm_read_cr3(&svm->vcpu);
+		hsave->save.cr3    = kvm_read_cr3(vcpu);
 
 	copy_vmcb_control_area(&hsave->control, &vmcb->control);
 
@@ -585,7 +586,7 @@ int nested_svm_vmrun(struct vcpu_svm *svm)
 	nested_svm_vmexit(svm);
 
 out:
-	kvm_vcpu_unmap(&svm->vcpu, &map, true);
+	kvm_vcpu_unmap(vcpu, &map, true);
 
 	return ret;
 }

From patchwork Wed Mar 24 17:50:05 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Krish Sadhukhan <krish.sadhukhan@oracle.com>
X-Patchwork-Id: 12162013
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,MSGID_FROM_MTA_HEADER,SPF_HELO_NONE,
	SPF_PASS,USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 5F11BC4345A
	for <kvm@archiver.kernel.org>; Wed, 24 Mar 2021 18:39:34 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 464DA61A29
	for <kvm@archiver.kernel.org>; Wed, 24 Mar 2021 18:39:34 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S237762AbhCXSjG (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Wed, 24 Mar 2021 14:39:06 -0400
Received: from userp2120.oracle.com ([156.151.31.85]:47706 "EHLO
        userp2120.oracle.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S237690AbhCXSiv (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 24 Mar 2021 14:38:51 -0400
Received: from pps.filterd (userp2120.oracle.com [127.0.0.1])
        by userp2120.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 12OIO2YQ185544;
        Wed, 24 Mar 2021 18:38:47 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : in-reply-to : references :
 content-transfer-encoding : content-type : mime-version;
 s=corp-2020-01-29; bh=InJGGPe9Kvj8qXOwTe7kukGFSVDP3QHi59gv0PQ8JME=;
 b=O3C6DBfuGh3E+tCIoKbfhngt1yciscrxmFq/AXcMKxRw8ku2uQsLSYy6rXeBjbesTEsP
 VoitazO0XWErae2qImg2aHYYoGG25dtQP4pjOTH3Mq9GjykptF802FBc5Pz/NDUdH/RY
 3L6BQEZrfJwxHv+UsE18n576qNEu2rmAt0AreOCQUa1P+mn8Jode1y5KzGnvK1h6LZAI
 epR5lYJiRi2rpLRDZkIBAIWnqJOxqq6Isx/UdWUuu9AAH+xF62+dlwtGmdYhwEoITBtL
 7Fd+u/CtLgN0bbyqvkiRkozmMONo4JzbkbfYFhFbwMo5DF+JN3STJc8DMbt4SfJpl09g 4g==
Received: from userp3020.oracle.com (userp3020.oracle.com [156.151.31.79])
        by userp2120.oracle.com with ESMTP id 37d9pn3sfj-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Wed, 24 Mar 2021 18:38:47 +0000
Received: from pps.filterd (userp3020.oracle.com [127.0.0.1])
        by userp3020.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 12OIQFPZ167563;
        Wed, 24 Mar 2021 18:38:47 GMT
Received: from nam04-dm6-obe.outbound.protection.outlook.com
 (mail-dm6nam08lp2046.outbound.protection.outlook.com [104.47.73.46])
        by userp3020.oracle.com with ESMTP id 37dtttpj74-2
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Wed, 24 Mar 2021 18:38:46 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=ml9JabzP6Kt8JaI9fh+FT5YNxz9AIVBnNMNggRkYDqRuPi/1NkATPYuxS8uk7oBpAKxhLIQZPqhoWiFt0ChHBq/t3w68ns4/ZEhUrKSYTNPbISMatF7M3QdrxDHC97wd005OIWoELUTtXx+XCFkkwH0x6OjR8oQoyHQ1aGAlKSv25bmUqk/joqJJFgyxHduQx3QjBdgbl/gZBoWb6L5Oa40y997IynkDkIlctImzdwcmwwfpALjymur+ATkvfjwxdG2gX4ADS6yqAaLHB2lDvpKUvUbD41e/3dHTZk+q7Rpqn7bgwn++pV1YEklJG3ctaMvZJe+UBC1UouWv5KH2yg==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=InJGGPe9Kvj8qXOwTe7kukGFSVDP3QHi59gv0PQ8JME=;
 b=h6J9QF5CZRMd/sROofp/FGkprNdLLrsgQTx67j13JNK2TwFH6ZmiTyWfR/XWw+rYnVxlvs/ORZbkxKIE4x/XK/5keuZcmHVqrZuMQ+dV/7bXGHTG2X+S4CgqyN7NCHqTHv/0FwMikasUADCwReAta6bDiA9nif7rlrblaaIVrzHIsHjKl8DGiHyu50Ur8wdU5Q4K4bU1eKO08MEu1B3Ni/pf8ZUv1+6Ana4Ei3VE9r20hh3t5Up4h9OLnEDoDhXJOhkZXQSh+IUj6Ptc6bmSb3bUmgi+WIDJFYeXxN24TUPmAW5xc07zbEPza1T+l+EPaaJIvdBXAE5fTP5vfIJIzQ==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=InJGGPe9Kvj8qXOwTe7kukGFSVDP3QHi59gv0PQ8JME=;
 b=Pz9CEOwyb/4RwMAz7fzhysGcV7l/o6D6XRlWrkDkcz6dcV80wOXNW5tmBOueXoQR22VJ3Iu+mjTeaSDPoyhFbsiiupYnWqduojjycgI5nQMnM7A/NvGoUvs5YAHo5ufMTudjsFv8yWNdMdGfKiJwuSP4BNlXRHm+Dl19AzRk9ik=
Authentication-Results: vger.kernel.org; dkim=none (message not signed)
 header.d=none;vger.kernel.org; dmarc=none action=none header.from=oracle.com;
Received: from SN6PR10MB3021.namprd10.prod.outlook.com (2603:10b6:805:cc::19)
 by SA2PR10MB4425.namprd10.prod.outlook.com (2603:10b6:806:11b::9) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3977.24; Wed, 24 Mar
 2021 18:38:45 +0000
Received: from SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::1871:3741:cc17:bcf7]) by SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::1871:3741:cc17:bcf7%7]) with mapi id 15.20.3955.025; Wed, 24 Mar 2021
 18:38:45 +0000
From: Krish Sadhukhan <krish.sadhukhan@oracle.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, jmattson@google.com, seanjc@google.com
Subject: [PATCH 4/5 v4] nSVM: Test addresses of MSR and IO permissions maps
Date: Wed, 24 Mar 2021 13:50:05 -0400
Message-Id: <20210324175006.75054-5-krish.sadhukhan@oracle.com>
X-Mailer: git-send-email 2.25.4
In-Reply-To: <20210324175006.75054-1-krish.sadhukhan@oracle.com>
References: <20210324175006.75054-1-krish.sadhukhan@oracle.com>
X-Originating-IP: [138.3.201.29]
X-ClientProxiedBy: BYAPR05CA0009.namprd05.prod.outlook.com
 (2603:10b6:a03:c0::22) To SN6PR10MB3021.namprd10.prod.outlook.com
 (2603:10b6:805:cc::19)
MIME-Version: 1.0
X-MS-Exchange-MessageSentRepresentingType: 1
Received: from ban25x6uut29.us.oracle.com (138.3.201.29) by
 BYAPR05CA0009.namprd05.prod.outlook.com (2603:10b6:a03:c0::22) with Microsoft
 SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3977.10 via Frontend
 Transport; Wed, 24 Mar 2021 18:38:44 +0000
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 15d212c8-6404-452b-1c65-08d8eef40e5a
X-MS-TrafficTypeDiagnostic: SA2PR10MB4425:
X-Microsoft-Antispam-PRVS: 
 <SA2PR10MB442577CBED365C94709C3EFB81639@SA2PR10MB4425.namprd10.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:4502;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 6cRCRzuIatbNaFssOOPr1TiwQ78tB3UXb5gi6s7kRE/kYbIm0adOHF5jYYTcl+jEWOV21nRjJtHy82UDxzG7daUheHK53+Ox4ZJaLRWbTTig12zh9eYQ48TivMfLW9p0DaEWvOdcjWC0E6nZ7QCflOtdFbG3pjgLxCHXSCrUnM/ST41qB5MZqNKsRR+NPatzzLdYz3oPbmILpyTBja/BgdR/TNihw1GchTbQ9tLb/ScWSett6VM8sSuOXKbwUgtQlCp9hj4Q94AFCHEYgTXm4dIlsyG3bVObD43HxSDDQb1QHJq3iInI1E6tM+BWmF/i445vF5bDzJv38EKsqfMHZPzzcObd7m2fbNetjr1tovF12e9JZA7nQ0SHGDMrnfgiCfYasO+ehey0JIaMI8XKcMb33vSuB+Va3QcVBCcGdsrAzmDEfGSXcWXemHCKhK2a+he1Cl4zDkJzcP0AN6bgx1fYlzHo+eh5OoZJZ9b+JuQhlFmN/GlFpZwRejxDy78rgjU612MFWs8KIh39J0qY9ixjh2/zuc/QEFZBO8mrVnxuuktRRIyIMDY+TPDKcSpMJ2JgGD0jq824RfSS9haLwUR8IRv6xTNIbeNVh8FReTM6Eaxh4HpB7zQEjNqn3OvqTaR/kuS/stvkK2iYndAXqQ==
X-Forefront-Antispam-Report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3021.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(136003)(346002)(376002)(366004)(396003)(39860400002)(38100700001)(52116002)(8936002)(7696005)(6666004)(36756003)(1076003)(5660300002)(6486002)(66946007)(66476007)(8676002)(66556008)(6916009)(2906002)(86362001)(26005)(956004)(186003)(478600001)(44832011)(16526019)(4326008)(2616005)(316002);DIR:OUT;SFP:1101;
X-MS-Exchange-AntiSpam-MessageData: 
 CgmJmYNu8rnhjyAM2NqFn3oaj0AR4PRD37iH/6CbYXelSZijY5v03fUAC9NDLbXzc+5BNDoj8iB2R3EQMJlZEzWdc5g50XhaARnT15Z1tVO1krDFIrZS3sR73h94pqaazX/DmyeLrnWSrCmbSH/vxbMxefy1s1++StrqFcNs9FtcRCUc1SNZniOlaP4EvzSIWWd88Vp1eZIqBcPl07IjJymbvyhXyl0CmOdD1OJ+grMWi2XQ0dGujIMV/15SMYo8x13d1EjVdlJkCh6TiLonkPP8E3nOJFpfXLmTB2NqA1/AK/kBAMln9cphc/dtPD+RHUq28/OHCYKtqQiml3QooZSBlnmv+R6Cfwia7BSIVy3TgIciODFup2cyKfm9E5T7unCdPx7Ba/8UoSOKSn9w6AxRDq50jMWM/rgc1wD5WX690+Mu36ZTzTg8gmHMDReEZELKtPz3ZmpgR6uusdFilCLUZkHsLRZ1ocZoNZSblM3DDXStrAVaC/vhfbgT5DZOvY/AfGXQV/WSObGgWuIf9v96RiwtrMnR19quYrpoDmf0J6t42d4S9/55blwz1JpB+7skonNpJpvPOzFFAVvCK4WqVY8YRykd5tUJTHmhg8rXiiJKLf/TBNDkRkBxRgEPgv+r4cd938mV9LTtFqTQztipSYIABPOUYdCasJcA8PyNc7zzE2oIwXRVcitNLUXfcA+cO3QUmHLm2dOH8t2p1NANrpAnwCJyxnnMWFjphyXV3nAkS49FzoFUrdSWtd/tk6eTOfa2yPJOz7fSxH+Ed0JcGBv+9DfEr/vwHzsn/nO9X6/GLHtejJh82Qaz2UsvDLWuQZHZB6I1GkId9adLYnfoizmBsU5dHlO5mLIADlyH/4Xb8LShTVWHSiRm0ikh02whnerd0edKNewT3yEmfMHTLXNDyS+ZePGguputC+CCwW9cSzyYRHg24JqXeKgYnbTcBwGbFXC4i8LudMQ26qlzNNPfp4G1nYL3hcp/9GZJGQejJ5wuaFhFU7jub1KPvBfcQcmxBlhLSWYDT2tsRrr09Rhb3neTevqP6UwBzPLMl7+raQtAYmpD+WSgvfcvXukadp5wjbtolwRHlwv2T75TovN4VlWhmIv4LwTIADHCbRar7VEwJkSfl1H2txIPFWYJL42MKapvNSjbSBa8TWvH3T63iuHkC5njJPz8D3mcQ8WpOdVoJZrZWhJHh5KM5zVf31pDSkczzXvk0dgsBIIq4czD2fPE/nmvRHfYZx4jWH6imo81mNBVcTAAJyTfj6h9U5I96ijCLzpVtNB/jxt664rP8lUESuzDaNEFxzb+d9pqo801ElmjTAu/TAhN
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 15d212c8-6404-452b-1c65-08d8eef40e5a
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3021.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 24 Mar 2021 18:38:45.5341
 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 bE55O/5en8sZBgLeEwG9YBhGRydqaPmVUQbekR1aHIMs1HsiGgfQMovWYHLqSkqVRhlu8caHGOXp3hiq5DvRj4iT3YjKX2sgLgtb99+92sg=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SA2PR10MB4425
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9933
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 adultscore=0
 malwarescore=0 spamscore=0
 mlxscore=0 phishscore=0 suspectscore=0 mlxlogscore=999 bulkscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2009150000
 definitions=main-2103240133
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9933
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0
 lowpriorityscore=0 phishscore=0
 mlxlogscore=999 priorityscore=1501 impostorscore=0 bulkscore=0 spamscore=0
 adultscore=0 clxscore=1015 malwarescore=0 mlxscore=0 suspectscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2009150000
 definitions=main-2103240133
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

According to section "Canonicalization and Consistency Checks" in APM vol 2,
the following guest state is illegal:

    "The MSR or IOIO intercept tables extend to a physical address that
     is greater than or equal to the maximum supported physical address."

Signed-off-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
---
 x86/svm_tests.c | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/x86/svm_tests.c b/x86/svm_tests.c
index 29a0b59..70442d2 100644
--- a/x86/svm_tests.c
+++ b/x86/svm_tests.c
@@ -2304,6 +2304,33 @@ static void test_dr(void)
 	vmcb->save.dr7 = dr_saved;
 }
 
+/*
+ * If the MSR or IOIO intercept table extends to a physical address that
+ * is greater than or equal to the maximum supported physical address, the
+ * guest state is illegal.
+ *
+ * [ APM vol 2]
+ */
+static void test_msrpm_iopm_bitmap_addrs(void)
+{
+	u64 addr_spill_beyond_ram =
+	    (u64)(((u64)1 << cpuid_maxphyaddr()) - 4096);
+
+	/* MSR bitmap address */
+	vmcb->control.intercept |= 1ULL << INTERCEPT_MSR_PROT;
+	vmcb->control.msrpm_base_pa = addr_spill_beyond_ram;
+	report(svm_vmrun() == SVM_EXIT_ERR, "Test MSRPM address: %lx",
+	    addr_spill_beyond_ram);
+	vmcb->control.intercept &= ~(1ULL << INTERCEPT_MSR_PROT);
+
+	/* MSR bitmap address */
+	vmcb->control.intercept |= 1ULL << INTERCEPT_IOIO_PROT;
+	vmcb->control.msrpm_base_pa = addr_spill_beyond_ram;
+	report(svm_vmrun() == SVM_EXIT_ERR, "Test IOPM address: %lx",
+	    addr_spill_beyond_ram);
+	vmcb->control.intercept &= ~(1ULL << INTERCEPT_IOIO_PROT);
+}
+
 static void svm_guest_state_test(void)
 {
 	test_set_guest(basic_guest_main);
@@ -2313,6 +2340,7 @@ static void svm_guest_state_test(void)
 	test_cr3();
 	test_cr4();
 	test_dr();
+	test_msrpm_iopm_bitmap_addrs();
 }
 
 

From patchwork Wed Mar 24 17:50:06 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Krish Sadhukhan <krish.sadhukhan@oracle.com>
X-Patchwork-Id: 12162015
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,MSGID_FROM_MTA_HEADER,SPF_HELO_NONE,
	SPF_PASS,USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 7245BC43446
	for <kvm@archiver.kernel.org>; Wed, 24 Mar 2021 18:39:34 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 5A97961A26
	for <kvm@archiver.kernel.org>; Wed, 24 Mar 2021 18:39:34 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S237777AbhCXSjI (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Wed, 24 Mar 2021 14:39:08 -0400
Received: from userp2120.oracle.com ([156.151.31.85]:47736 "EHLO
        userp2120.oracle.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S237693AbhCXSiz (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 24 Mar 2021 14:38:55 -0400
Received: from pps.filterd (userp2120.oracle.com [127.0.0.1])
        by userp2120.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 12OIOSX3185704;
        Wed, 24 Mar 2021 18:38:51 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : in-reply-to : references :
 content-transfer-encoding : content-type : mime-version;
 s=corp-2020-01-29; bh=FCHR/kvHlyKbKvUjMwXY42SxTz0rqXup5nfnNHN3YTM=;
 b=c4blUcuuSmQz15sANkHEu0foKNP7C5WJLmcC6oNN6WFKq6kFXuf5ClNyzzE8IudXJLB4
 xZqh8QUHGWR3IaMCIBLgNdU20IXjk3+XgE1m00kDd3QXLbM6H+DUsPSe/x7F7fPUnAUN
 OhCr6Is90uZS8bmsbJfMNvxpB5g10ti4Ocaj4RqAqvjz7k5GAtLQsvDp0YZ+cmEb0K+d
 8WaGMa5x9fiWHNZI8NuDU60k8SGfXz0VFZbF4L9xPIGcHdRX9NZL3/6enWC50ZfXZvQl
 Tj/gMpfEinXJl/GyIK+80QSXaudnf6DFkRHdJNEDrOd1oLO7S3wFKRIwT26HmQecaLgx Lg==
Received: from userp3030.oracle.com (userp3030.oracle.com [156.151.31.80])
        by userp2120.oracle.com with ESMTP id 37d9pn3sfp-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Wed, 24 Mar 2021 18:38:51 +0000
Received: from pps.filterd (userp3030.oracle.com [127.0.0.1])
        by userp3030.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 12OIOsKC053055;
        Wed, 24 Mar 2021 18:38:51 GMT
Received: from nam04-sn1-obe.outbound.protection.outlook.com
 (mail-sn1nam04lp2051.outbound.protection.outlook.com [104.47.44.51])
        by userp3030.oracle.com with ESMTP id 37du00675x-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Wed, 24 Mar 2021 18:38:50 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=TWDrI4X3NbmReJegVlKIdxRbSlT/yrxXa71D0DEj+JLwV1wKPpWWEKCXCTpAuQ0j8orfLELV39qLTBtekcwgaoyEjI7fHSYpHbBdK/eXYyknxSgU1/SpxpXPNjgJavOklkj98blWb1fuGKrXJPOl28BPIP5iA8y/zzDzVYXzwfK21w/HGOLtGqbQ8qQFtUw6eDDiSwlSb2YvCUheVNHrTR3/HopPx951khWD/FfuN5+pNjbD6lHyL+BcDgC4jleyjRY2JoYjGSZ2RvuK7epP4s0DoH8PLX4Bd3U4f2W8WKuJaOrJZtByk1nlJLsbDSGcV4xNEcvlxFA/HkDDJHcnaw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=FCHR/kvHlyKbKvUjMwXY42SxTz0rqXup5nfnNHN3YTM=;
 b=R03Sw3gqdh7MR2eGkc1ZR2VcK+WNnurx9ok4vflqF7UI6AN4QD51ybKoALFr4bEs15GSbIpEX96jVn2FHXQA+LlIpv/C+fHCwcZ4+6z9LpJWKYTUnRLKuiRR1dqkgwxDYsyv1xvw+1rCA06+5aErAH6OW/7pdk/iiqEvDjXIqz858ID27I1xQXoKhIT/ahyfEXCJQvWmLphbnpjbitLyI/h/6fr9lZ0fP9mZ6Gqu9b3Xuidg+5MQ9kXYCj/6KF0U5TKi0XvNneK52enyHI2hSr6WXi2CSrq0fe+fqjtpC09a09HIQDVQabNl7fKX89RjM4e5Skuoxh8abNGiAelNRg==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=FCHR/kvHlyKbKvUjMwXY42SxTz0rqXup5nfnNHN3YTM=;
 b=XrZ5LwKvVJw3KYGqE/h5RzRy+oIdCxjTTj/UcAbmTs1NLvb5tDyCvHhnhfYpDhUGiHcAvjtpsJizR1DvO9sWCB599alJ2vDvwM4EHYbpLLPCWeBH0j+soyxhseHrjcORXtgeeWcL2JwgoLwZWqwh6QtKsmWecFa/msL88J4C/BY=
Authentication-Results: vger.kernel.org; dkim=none (message not signed)
 header.d=none;vger.kernel.org; dmarc=none action=none header.from=oracle.com;
Received: from SN6PR10MB3021.namprd10.prod.outlook.com (2603:10b6:805:cc::19)
 by SN6PR10MB2991.namprd10.prod.outlook.com (2603:10b6:805:d3::12) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3955.18; Wed, 24 Mar
 2021 18:38:48 +0000
Received: from SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::1871:3741:cc17:bcf7]) by SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::1871:3741:cc17:bcf7%7]) with mapi id 15.20.3955.025; Wed, 24 Mar 2021
 18:38:46 +0000
From: Krish Sadhukhan <krish.sadhukhan@oracle.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, jmattson@google.com, seanjc@google.com
Subject: [PATCH 5/5 v4] SVM: Use ALIGN macro when aligning 'io_bitmap_area'
Date: Wed, 24 Mar 2021 13:50:06 -0400
Message-Id: <20210324175006.75054-6-krish.sadhukhan@oracle.com>
X-Mailer: git-send-email 2.25.4
In-Reply-To: <20210324175006.75054-1-krish.sadhukhan@oracle.com>
References: <20210324175006.75054-1-krish.sadhukhan@oracle.com>
X-Originating-IP: [138.3.201.29]
X-ClientProxiedBy: BYAPR05CA0009.namprd05.prod.outlook.com
 (2603:10b6:a03:c0::22) To SN6PR10MB3021.namprd10.prod.outlook.com
 (2603:10b6:805:cc::19)
MIME-Version: 1.0
X-MS-Exchange-MessageSentRepresentingType: 1
Received: from ban25x6uut29.us.oracle.com (138.3.201.29) by
 BYAPR05CA0009.namprd05.prod.outlook.com (2603:10b6:a03:c0::22) with Microsoft
 SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3977.10 via Frontend
 Transport; Wed, 24 Mar 2021 18:38:45 +0000
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: feb939d8-3219-464c-c9d8-08d8eef40f12
X-MS-TrafficTypeDiagnostic: SN6PR10MB2991:
X-Microsoft-Antispam-PRVS: 
 <SN6PR10MB299131EAE546A5DC66DCBAE081639@SN6PR10MB2991.namprd10.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:2958;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 8mtOZk/cVtvtlF430l66SF+ioI8qZ6dYgjeTefdITFcFwTNzASBOXh3Nl92KD9zMIfPszxNWA+JuyP3b3FAU6Pbo2Z0/q56FQAYw0H8JF2BDgc7WKvo0D6OOEs8FyspXKNSGzNY4kmgSoH9LhSrC4pZSEDh2ZNEAqF09tGWqRuKwSljH/GX2X1lHRbzu29GsPg2gC/c7zkZ5zUhAUu93JMlS2TCRQ3OAcb5oC+iDIpK3jFQy1VeXi1+vGgW355OwvbJ66wSNB2aku2MOBc8Oz5lDCwI/DIy0YGK2ft/uRAThAUeJR/mX9p27j/GsyvizAWsrkUbmXUZZyX1nEaWTPoMaxYD6wqcV37jkT3QDNEoonqu5ei50jscm8y0Y2Kg2LfvKDTQk7PPYDmSix6o+aqMPPWt8BXEH0yJah06775hOqFf1ht+CVNYXK4gKH6wnge1nEtBHPRV48SFPyq8hL3+mhkUoQP0nZta9WRqPZbJ8Q8FdFIJCyC0pf55zIP/mQsnMVA7L4yKJbBf4tmM4aHh3++2F889X3KKiExNZC5GICTuP4JXeoD1WewBwA1DZnzkzyWexGdg/zLYeYAB0aoLym5m0Nmy0cyBg8CkYju8rUtpGpwcA22IT3WwgiwfdJPAtKBPN9L+VWQvRchYM2w==
X-Forefront-Antispam-Report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3021.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(376002)(136003)(346002)(39860400002)(396003)(366004)(52116002)(6486002)(38100700001)(4744005)(36756003)(6666004)(478600001)(1076003)(7696005)(6916009)(83380400001)(4326008)(956004)(66476007)(44832011)(8676002)(2616005)(186003)(26005)(2906002)(8936002)(16526019)(316002)(66946007)(66556008)(86362001)(5660300002);DIR:OUT;SFP:1101;
X-MS-Exchange-AntiSpam-MessageData: 
 VHBAbqP6+eNsOX2gsiVSw2KA5z/XqWFy2qHYPVjvsY/KPOYyH2w84u0tRgsm9RXSpETXAzSrLn3OzgcDHwCgbVUcocgMUyepwvUCi0tW5oGpbYJRUCzXTY+m2TGrhSv/T2YbrL/TJAlF9lgj9ZFgti/Fq02BdtkOcs71aQSdIYkOjkvrKQnL1NNZ32WSvOLbZ+x9IgXOf69n/EyyyBSPeg6q+Ifmcu06CjOFjge/92B5UpffALPIabNXACxArfPloPcqgMtvrz/2R5sCWzClo0kLtWR3AbsJIt5sa3L+IPCYbt2IBHbUjSMVlJl/YguTXg23PDfBm3aCCguG/vRu+pE2jQms1PCzP9/ItgIuzsJnd01UwA0qDndvy4wah+tRSdf+UaEFwLf5lDUKwOzlOFmDr89JWdRM0OG/VzOrcaxAEttnDntRV9oKv4cLh1Y6M/WHrYnZx2Sgg/xh2+Je0Kih2AjJpryhxqdUqoOEjztyF585jkL2Do4MaYitfYSQtyKf6VLfbOv0zU3FFsNPwKHSkaJSM1BNdH9pXXrXgFXGfkioTsWGtzTG02yjsehKB1kWbt+McYprvqdU3z6LEMvCQH2tz0UjbcbY03RoXTCqvFIIPSSs2FeXvT/H7Cy1QbGTsDQIciSLp6OB+K687/KfKvq/p1W8u6gkdpdWb5qhDco2yLDmwYXUdTpANT8TO34gTeL8+/8OqhQAgf40G2vmDN8ILRggP6eGPrAVDwDo1fDzPJt0tL1VL3+eWsToRQeqK29MizJyNIA680xZ8qkYVN0Llz/qnkfeciZUzbi+w7IuPQLP9aRGiaUfl/P+I82aS9iXF112xZxEntQ6pMRugYe2fVeCUGVJqmSdNJ6eqsgI5Hr5RxbwjBU7ME7gUV388Mf7/rRrC2OJn//ogEfajdcpZRLD7g2M0Ce7x40evkgTecj8MYVy6PIT1I9+Lb315y52tS+QkTpQtxuXiGAjs+8K828pywM/tvEpws78mgLVmcSHgctb3wBO3cyYaQ87yx9FNm/NJmNku6W3Q7X4lvVRZ03CALYmYWuOLYT5g0hFc4auzf4ZbXj2vNsx7+Z7kqYBcFRxYUaoJl0CMtsdI8LAUs1JNiDY40K5G6p2Yu48gPpKyqVhtH8PFdW3E94Uf9DPFHtn3Du2UlDSS5XMVUdYIg8n88hhBvz1ILXQbidkY3pVyEWG9uYlLBjTmqFsxg23FnhtISLWZaaZ6cCS2PjYPQ52aS9hJyy4KUi24GKLZrMZn32Elgis1fgRzIuG5CWqZ5oQQ/o1f2jOEVZ2wSatRxsGLtkOxN5Ju5G6IVmb4f4xrQyRYJqAOhLR
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 feb939d8-3219-464c-c9d8-08d8eef40f12
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3021.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 24 Mar 2021 18:38:46.7694
 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 +wbEFnzWaysjIVTCOb1RkDiQa8B22mP3X0SXuAQIDFr8XNbXqpkLfM2LeHeoo8vwv0tvxcumq6dCCPm5tYZFkgSrJ6smnPpb8m8F5oFC7H8=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SN6PR10MB2991
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9933
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 adultscore=0
 bulkscore=0 phishscore=0
 mlxlogscore=999 suspectscore=0 spamscore=0 malwarescore=0 mlxscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2009150000
 definitions=main-2103240133
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9933
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0
 lowpriorityscore=0 phishscore=0
 mlxlogscore=999 priorityscore=1501 impostorscore=0 bulkscore=0 spamscore=0
 adultscore=0 clxscore=1015 malwarescore=0 mlxscore=0 suspectscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2009150000
 definitions=main-2103240133
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Since the macro is available and we already use it for MSR bitmap table, use
it for aligning IO bitmap table also.

Signed-off-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
---
 x86/svm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/svm.c b/x86/svm.c
index a1808c7..846cf2a 100644
--- a/x86/svm.c
+++ b/x86/svm.c
@@ -298,7 +298,7 @@ static void setup_svm(void)
 	wrmsr(MSR_EFER, rdmsr(MSR_EFER) | EFER_SVME);
 	wrmsr(MSR_EFER, rdmsr(MSR_EFER) | EFER_NX);
 
-	io_bitmap = (void *) (((ulong)io_bitmap_area + 4095) & ~4095);
+	io_bitmap = (void *) ALIGN((ulong)io_bitmap_area, PAGE_SIZE);
 
 	msr_bitmap = (void *) ALIGN((ulong)msr_bitmap_area, PAGE_SIZE);
 
